### PR TITLE
Add Quality parameter to BrowserFileExtensions.RequestImageFileAsync

### PR DESCRIPTION
**Pull request by chrfin**
**Add Quality parameter to BrowserFileExtensions.RequestImageFileAsync**
This adds a quality parameter to the `BrowserFileExtensions.RequestImageFileAsync` method to choose the quality of the resized image.

Addresses #32769

### CODE REVIEW

**Comment by mkArtakMSFT** [comment]
@TanayParikh would you mind reviewing this. Looks like @javiercn has a lot on his plate and didn't get chance to review this.

File: src/Components/Web/src/Forms/InputFile/BrowserFileExtensions.cs [hunk 1]
```diff
@@ -26,12 +26,13 @@ public static class BrowserFileExtensions
         /// <param name="format">The new image format.</param>
         /// <param name="maxWidth">The maximum image width.</param>
         /// <param name="maxHeight">The maximum image height</param>
+        /// <param name="quality">The quality of the image</param>
         /// <returns>A <see cref="ValueTask"/> representing the completion of the operation.</returns>
-        public static ValueTask<IBrowserFile> RequestImageFileAsync(this IBrowserFile browserFile, string format, int maxWidth, int maxHeight)
+        public static ValueTask<IBrowserFile> RequestImageFileAsync(this IBrowserFile browserFile, string format, int maxWidth, int maxHeight, double? quality = null)
```
Comment (pranavkm) [inline src/Components/Web/src/Forms/InputFile/BrowserFileExtensions.cs:31]: You'll need to add a new overload that accepts the quality parameter. Adding an additional parameter to an existing method is a binary breaking change.

File: src/Components/Web/src/Forms/InputFile/BrowserFileExtensions.cs [hunk 2]
```diff
@@ -10,7 +10,7 @@ namespace Microsoft.AspNetCore.Components.Forms
     /// Contains helper methods for <see cref="IBrowserFile"/>.
     /// </summary>
     public static class BrowserFileExtensions
-    {
+    {
```
Comment (SteveSandersonMS) [inline src/Components/Web/src/Forms/InputFile/BrowserFileExtensions.cs:13]:
```suggestion
    {
```

**Comment by SteveSandersonMS** [comment]
Other than the [whitespace typo](https://github.com/dotnet/aspnetcore/pull/32772/files/ae431b44e618e0d322519f2819651723b70934f1#diff-d2f9e57fae8b7691518b32266f6fc0ed74755d3f0e07016733c29a8f2b331856), this looks good to me, but I'll leave it for @pranavkm to review too.

File: src/Components/Web/src/Forms/InputFile/BrowserFileExtensions.cs [hunk 3]
```diff
@@ -28,10 +28,33 @@ public static class BrowserFileExtensions
...
+        /// </summary>
+        /// <remarks>
+        /// The image will be scaled to fit the specified dimensions while preserving the original aspect ratio.
+        /// The quality parameter is not supported on all browsers, especially mobile browsers.
+        /// </remarks>
+        /// <param name="browserFile">The <see cref="IBrowserFile"/> to convert to a new image file.</param>
+        /// <param name="format">The new image format.</param>
+        /// <param name="maxWidth">The maximum image width.</param>
+        /// <param name="maxHeight">The maximum image height</param>
+        /// <param name="quality">The quality of the image</param>
```
Comment (SteveSandersonMS) [inline src/Components/Web/src/Forms/InputFile/BrowserFileExtensions.cs:51]: Could we extend the comment to give a hint about what is the reasonable range of values? For example, does it go zero to one, or zero to one hundred (a percentage), or zero to infinity, or do the values have some other meaning? What nonempty value is equivalent to the default if you leave it unspecified?

Comment (chrfin) [reply src/Components/Web/src/Forms/InputFile/BrowserFileExtensions.cs]: Good idea - I added the text from MDN.

File: src/Components/Web/src/Forms/InputFile/BrowserFileExtensions.cs [hunk 4]
```diff
@@ -25,13 +25,40 @@ public static class BrowserFileExtensions
...
+        /// <remarks>
+        /// The image will be scaled to fit the specified dimensions while preserving the original aspect ratio.
+        /// The quality parameter is not supported on all browsers, especially mobile browsers.
+        /// </remarks>
+        /// <param name="browserFile">The <see cref="IBrowserFile"/> to convert to a new image file.</param>
+        /// <param name="format">The new image format.</param>
+        /// <param name="maxWidth">The maximum image width.</param>
+        /// <param name="maxHeight">The maximum image height.</param>
+        /// <param name="quality">
+        /// A Number between 0 and 1 indicating the image quality to be used when creating images using file formats that support
```
Comment (SteveSandersonMS) [inline src/Components/Web/src/Forms/InputFile/BrowserFileExtensions.cs:52]:
```suggestion
        /// A number between 0 and 1 indicating the image quality to be used when creating images using file formats that support 
```

**Review by pranavkm — COMMENTED** [review]
Do we already have an end-to-end test for RequestImageFileAsync? If so, could we add another one that covers passing the new parameter?

**Comment by TanayParikh** [comment]
> Do we already have an end-to-end test for RequestImageFileAsync? If so, could we add another one that covers passing the new parameter?

Yes we should be able to create something similar to the test below using the [Input File test component](https://github.com/dotnet/aspnetcore/blob/main/src/Components/test/testassets/BasicTestApp/FormsTest/InputFileComponent.razor).

https://github.com/dotnet/aspnetcore/blob/2656c454d6c3f5e3d62e53e6ceb37e0705bc7fab/src/Components/test/E2ETest/Tests/InputFileTest.cs#L127-L158

**Comment by pranavkm** [comment]
Thanks for your PR. We'd be happy to revisit this once you're able to address the PR feedback we provided.

**Comment by chrfin** [comment]
@pranavkm Did I miss something? I was under the impression I did address every feedback provided...

**Comment by chrfin** [comment]
@SteveSandersonMS Ah, ok. I interpreted the "we" from @TanayParikh and the fact that the example does not use the old `RequestImageFileAsync` as if this is something you need to do.

I will investigate how I could run those tests on my side and then try to implement a test for both overloads of `RequestImageFileAsync` sometime this week 🤓.

**Pull request closed without merging**